src/views/editConnection.ts

```diff
@@ -197,14 +198,18 @@ export const createEditConnection = (m, pubsub, settings: ISettings) => {
                                                             ],
                                                         },
                                                     ],
-                                                },
-                                                res => {
-                                                    if (res && res[0]) {
-                                                        m.startComputation()
-                                                        image(res[0])
-                                                        m.endComputation()
-                                                    }
                                                 }
+                                                // res => {
```

Comment:
I couldn't find anything in the mithril documentation about the remote function using a callback function like this. It did not pass type checking, so I commented it out instead of deleting it.

Reply:
dialog.showOpenDialog is an electron api. This function has changed to return a promise instead of using a callback. See: 
https://www.electronjs.org/docs/api/dialog

I think this should work:
`dialog
  .showOpenDialog(
      remote.getCurrentWindow(),
      {
          filters: [
              {
                  name: 'images',
                  extensions: [
                      'png',
                      'jpg',
                  ],
              },
          ],
      }
  )
  .then(res => {
      if (
          res &&
          !res.canceled &&
          res.filePaths &&
          res.filePaths[0]
      ) {
          m.startComputation()
          image(res.filePaths[0])
          m.endComputation()
      }
  })`